tools/datagen/src/bin/datagen.rs

```diff
@@ -370,6 +373,14 @@ fn export_cldr<'data>(
 
     let keys = get_all_cldr_keys();
 
+    let keys = if let Some(allowed_keys) = allowed_keys {
+        keys.into_iter()
+            .filter(|k| allowed_keys.contains(&*k.to_string()))
+            .collect()
```

Comment:
Praise: I had thought about doing this in the other direction (convert `allowed_keys` into `ResourceKey` rather than `keys` into `String`), but the way you did it is much easier.